### PR TITLE
Feat project basic architecture

### PR DESCRIPTION
**Pull request by danielfigueiredo**
@tyllrb @mattmcfad 
This is what I've gotten so far. I look forward to hear feedback from you.
Currently we're supporting the following:
- Environment configuration
- Server startup depending on database connection
- Unit and API tests
- Consolidated coverage report
- Linting validation with jshint and jscs
- Watcher for automatically run tests

Just to make an example I created a simple user api and wrote one unit and one api test for it.
This example should be deleted after we define the data model and have a real api to build.

The architecture now basically have a layer called `services` (`app/services/`) that holds our business algorithms. I think everything should be a service, a service can call another service. In case we start having big transactions that involves multiples services we may think about an orchestrator layer that organizes and executes multiples services.
Now basically the layers are mostly stacked, we have one vertical layer (cross layers) that is our utility `lang.js` file. The execution flow for a request is:
 `routes` -> `services` -> `providers` -> `database` <---
This means that a `service` should never require a `route` file and `provider` should never require a `service` file.

The test folder structure resembles the same architecture. In other words to find the tests for files located under:
`app/routes/*.js`
you can go to
`test/src/routes/*.js`

That's all I think =]

### CODE REVIEW

File: test/src/api/global.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+sinon.stub(configResource, 'getDBConnectionURI', () =>
+  configResource.get('MONGO_TEST_URI') + configResource.get('MONGO_TEST_DB')
+);
+
+/**
+ * Start up the server, clean up the database before running tests.
+ */
+before(function(done) {
+  this.timeout(5000);
```
Comment (mattmcfad) [inline test/src/api/global.js:23]: is this timeout a workaround for something?

Comment (danielfigueiredo) [inline test/src/api/global.js:23]: That is a timeout to control the amount of time the server needs to start. This is a global before that starts the server before running tests.
I lowered it down to 1 second and still works on my machine, but may vary depending on the machine.

File: app/apis/database-api.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const configResource = require('../resources/config-resource');
+const Rx = require('rxjs');
+
+function createConnection() {
+  return Rx.Observable.create(observer => {
+    mongoose.connect(configResource.getDBConnectionURI());
```
Comment (mattmcfad) [inline app/apis/database-api.js:9]: out of curiosity, why `.connect()` vrs `.createConnection()` ?

Comment (danielfigueiredo) [inline app/apis/database-api.js:9]: no idea, maybe better make both the same name replacing line 7 with connect
=]

**Comment by danielfigueiredo** [comment]
Guys I'm going to merge this because the other branch I branched off this is getting huge
We can always go back and change stuff =]

**Pull request merged**